src/app/components/tables/patient-table/patient-table.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, ViewChild, Input, OnInit } from '@angular/core'
+import { DataSource } from '@angular/cdk'
+import { MdPaginator } from '@angular/material'
+import { Observable } from 'rxjs/Observable'
+import 'rxjs/add/operator/startWith'
+import 'rxjs/add/observable/merge'
+import 'rxjs/add/operator/map'
+
+import { Subject } from '../../../shared/store/subject/subject.model'
+
+export class PatientDataSource extends DataSource<any> {
+  mydata: Subject[]
+
+  constructor (private subjects: Observable<Subject[]>, private _paginator: MdPaginator) {
+    super()
+    this.subjects.subscribe(res => this.mydata = res)
```

Comment:
`res` is short for response, which is used when there's a data response from an http call, use `d` or `x` for generic data.